frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -133,6 +134,18 @@ class Worksheet extends React.Component {
             success: function(info) {
                 info['date_created'] = addUTCTimeZone(info['date_created']);
                 info['date_last_modified'] = addUTCTimeZone(info['date_last_modified']);
+                // Use the cache to update directive block if we are in the partial update mode
+                if (props.skipSearchRefresh) {
```

Comment:
Actually, it's considered not the best practice to store cache in state. Can we use something like `useSWR` (https://swr.vercel.app/getting-started) for the cache instead?